utils/build-apple-framework.sh

```diff
@@ -92,28 +100,18 @@ function build_apple_framework {
   fi
 }
 
-# Accepts an array of frameworks and will place all of
-# the architectures into the first one in the list
+# Merges compiled frameworks and combines all of
+# the architectures into the single xcframework
 function create_universal_framework {
   cd ./destroot/Library/Frameworks || exit 1
 
-  local platforms=("$@")
-
-  echo "Creating universal framework for platforms: ${platforms[*]}"
-
-  for i in "${!platforms[@]}"; do
-    platforms[$i]="${platforms[$i]}/hermes.framework/hermes"
-  done
-
-  lipo -create -output "${platforms[0]}" "${platforms[@]}"
+  echo "Creating universal framework for Apple Platforms"
 
-  # Once all was linked into a single framework, clean destroot
-  # from unused frameworks
-  for platform in "${@:2}"; do
-    rm -r "$platform"
-  done
+  xcodebuild -create-xcframework -framework iphoneos/hermes.framework -framework iphonesimulator/hermes.framework -framework macosx/hermes.framework -output iphoneos/hermes.xcframework
```

Comment:
The output should not assume the framework is just for iOS here, it should theoretically also be able to contain the macOS slices, no?

Reply:
Yes. Also, it should also get the platforms from the `platforms`, like the `lipo` did, without hardcoding them. I just think it's nicer and doesn't require manual `rm` of every folder, like it's currently done on line 112-114.